x/nft/codec.proto

```diff
@@ -32,10 +32,23 @@ message AddApprovalMsg {
     bytes address = 2;
     string action = 3;
     ApprovalOptions options = 4 [(gogoproto.nullable) = false];
+    Type t = 5;
 }
 
 message RemoveApprovalMsg {
     bytes id = 1;
     bytes address = 2;
     string action = 3;
+    Type t = 4;
+}
+
+enum Type {
+    Username = 0;
+    Ticker = 1;
+    Blockchain = 3;
+}
+
+enum Action {
```

Comment:
pp: I am sorry to nitpicking on this but I don't see the benefit in having an action eum type in `nft` package protobuf. Subtypes will have custom actions defined in their subpackages while `nft` defines the base actions valid for all types. The type consts and validation map can be written easily by hand. We should not have orphan types in protobuf defined that we don't use in IO nor persistence IMHO. That would be confusing to anybody reading the file.

Reply:
Since we decided to go that way with Actions it has to be defined in both base and implementations as an enum.
What I can do is move this codec to "base" package where the handler for approvals is, then it's going to be less confusing.

Reply:
Sorry, I thought this was a proposal. We can keep your enum for this PR but let's revisit this when we implement fine grained permissions into blockchain nft.